dowhy/causal_refuters/linear_sensitivity_analysis.py

```diff
@@ -0,0 +1,435 @@
+
+    
+    def plot_t(self,r2tu_w, r2yu_tw):
+        """
+        Computes the contours, threshold line and bounds for plotting t.
+        Contour lines (z - axis) correspond to the adjusted t values for different values of r2tu_w (x) and r2yu_tw (y).
+        :param r2tu_w: hypothetical partial R^2 of confounder with treatment(x - axis)
+        :param r2yu_tw: hypothetical partial R^2 of confounder with outcome(y - axis)
+
+        :returns:
+        contour_values : values of contour lines for the plot
+        critical_t : threshold point 
+        t_bounds : t-value for unobserved confounders (bias adjusted t values)
+        """
+
+        t_alpha_df_1 = t.ppf(self.significance_level / 2 , self.degree_of_freedom - 1) # t-value threshold with alpha significance level and dof-1 degrees of freedom
+        critical_t = abs(t_alpha_df_1) * np.sign(self.t_stats)
+
+        contour_values = []
+        for x in r2tu_w:
+            contour = []
+            for y in r2yu_tw:
+                benchmarking_results = self.compute_bias_adjusted(r2tu_w = x, r2yu_tw = y)
+                t_value = benchmarking_results['bias_adjusted_t']
+                contour.append(t_value)
+            contour_values.append(contour)
+
+        t_bounds = self.bias_adjusted_t
+        return contour_values, critical_t, t_bounds
+
+    
+    def plot(self, sensitivity_variable = 'estimate', critical_estimate = 0, critical_t = 2, contour_color = "blue", threshold_color = "red", x_limit = 0.8, y_limit = 0.8 ):
+        """
+        The horizontal axis shows hypothetical values of the partial R2 of unobserved confounder(s) with the treatment
+        The vertical axis shows hypothetical values of the partial R2 of unobserved confounder(s) with the outcome.
+        The contour levels represent adjusted t-values or estimates for unobserved confounders with these hypothetical partialR2 values.
+        We also plot bounds on the partial R^2 of the unobserved confounders obtained from observed covariates
+
+        :param sensitivity_variable: "estimate" or "t-value"
+        :param critical_estimate: threshold line to represent the contour corresponding to estimate, (default = 0)
+        :param critical_t: threshold line to represent the contour corresponding to t-value, (default = 2 - usual approx value for 95% CI)
+        :param x_limit: plot's maximum x_axis value
+        :param y_limit: plot's minimum y_axis value
+        :param contour_color: color of contour line(default = blue)
+        :param threshold_color: color of threshold line(default = red)
+        """
+
+        #Plotting the contour plot
+        fig, ax = plt.subplots(1, 1, figsize = (7,7))
+        ax.set_title("Sensitivity contour plot of %s"  %sensitivity_variable)
+        ax.set_xlabel("Partial R^2 of confounder with treatment")
+        ax.set_ylabel("Partial R^2 of confounder with outcome")
+
+        for i in range(len(self.r2tu_w)):

```

Comment:
I didn't understand the logic of this code. Why 0.95 and not 0.99?

Reply:
And why is 0.8 default?

Reply:
Yes 0.99 seems right.
Beyond 0.8, we mostly deal with extreme scenarios and the plot also looks distorted for such cases. therefore for general cases, the limits were set to 0.8 and above that we can set it to 0.99

Reply:
okay sounds good.